src/pages/Dashboard/Markets/SupplyMarket.tsx

```diff
@@ -114,14 +116,20 @@ const SupplyMarket: React.FC = () => {
   const userMarketInfo = useUserMarketInfo({ account });
   const [confirmCollateral, setConfirmCollateral] = useState<Asset | undefined>(undefined);
   const { mutate: enterMarkets } = useEnterMarkets({
-    onSuccess: () => setConfirmCollateral(undefined),
+    onSuccess: () => {
+      queryClient.invalidateQueries(FunctionKey.GET_ASSETS_IN_ACCOUNT);
```

Comment:
I'd suggest invalidating the query directly from the `useEnterMarkets` hook, this way the hook becomes also responsible for handling cache (so if we import this mutation from somewhere else, cache will be handled out of the box).
This pattern will become particularly interesting if we get into optimistic updates, or even just updating the cache on the fly after a successful mutation.

Reply:
That makes sense. It brings me back to a thought I had that all options should be included in the hook because it seems like all options should belong in the hook with the exception of acceptable overrides

Reply:
So the way I see it is that we should assign all the options related to the hook inside it directly, but keep the option of overriding them as you mentioned.
I wouldn't filter what options can be overridden or not, since some very specific cases will sometimes need to override options that were defined in the hook directly.

Reply:
In that case I think we should have a merge strategy so that everything gets merged as desired. Otherwise we'll have lots of duplicated code managing merging options

Reply:
I don't know if we can generalize the merging strategy though, as some components won't extend the options at all and just pass the ones they receive, while some will have custom behaviours like in your case:
```typescript
const useCustomQuery = (options?: Options) => (
  useQuery(
    FunctionKey.REPAY_VAI,
    fetchFn,
    {
      ...(options || {})
      onSuccess: (...props) => {
        // Invalid queries here

        if (options?.onSucess) {
          onSuccess(...props);
        }
      }
    },
  )
);
```